apps/backend/src/tests/response_api.test.ts

```diff
@@ -0,0 +1,555 @@
+      .set('Authorization', `Bearer ${token}`)
+      .send({ time_created: new Date() });
+
+    expect(response4.status).toBe(400);
+    expect(response4.body.error).toBe(
+      'Content and/or status value is invalid.',
+    );
+
+    const responseInDbAfterUpdate = await testHelpers.getSingleResponse(
+      ANTONINAS_RESPONSE_ID,
+    );
+    expect(responseInDbAfterUpdate).toEqual(responseInDbAfterUpdate);
+  });
+
+  test('Able to change to status of response to ACCEPTED', async () => {
+    const initialResponseInDb = await testHelpers.getSingleResponse(
+      ANTONINAS_RESPONSE_ID,
+    );
+    expect(initialResponseInDb.status).toBe('PENDING');
+
+    // Why can we not use type of `updateData` to be `UpdateRequestData` ?
+    // TS still expects `status` field to be of type `number`
+    const updateData = {
+      status: 'ACCEPTED',
+    };
+
+    const httpResponse: Response = await api
+      .put(`/api/responses/${ANTONINAS_RESPONSE_ID}`)
+      .set('Authorization', `Bearer ${token}`)
+      .send(updateData);
+
+    expect(httpResponse.status).toBe(200);
+    expect(httpResponse.body.id).toBe(ANTONINAS_RESPONSE_ID);
+    expect(httpResponse.body.status).toBe('ACCEPTED');
+    expect(httpResponse.body.content).toBe(initialResponseInDb.content);
+
+    const finalResponseInDb = await testHelpers.getSingleResponse(
+      ANTONINAS_RESPONSE_ID,
+    );
+
+    expect(finalResponseInDb.id).toBe(initialResponseInDb.id);
+    expect(finalResponseInDb.request_id).toBe(initialResponseInDb.request_id);
+    expect(finalResponseInDb.user_id).toBe(initialResponseInDb.user_id);
+    expect(finalResponseInDb.content).toBe(initialResponseInDb.content);
+    expect(finalResponseInDb.status).toBe('ACCEPTED');
+  });
+
+  test('Able to change the content of response', async () => {
+    const initialResponseInDb = await testHelpers.getSingleResponse(
+      ANTONINAS_RESPONSE_ID,
+    );
+
+    const updateData = {
+      content: 'test content',
+    };
+
+    const httpResponse: Response = await api
+      .put(`/api/responses/${ANTONINAS_RESPONSE_ID}`)
+      .set('Authorization', `Bearer ${token}`)
+      .send(updateData);
+
+    expect(httpResponse.status).toBe(200);
+    expect(httpResponse.body.id).toBe(ANTONINAS_RESPONSE_ID);
+    expect(httpResponse.body.status).toBe('PENDING');
+    expect(httpResponse.body.content).toBe(updateData.content);
+
+    const finalResponseInDb = await testHelpers.getSingleResponse(
+      ANTONINAS_RESPONSE_ID,
+    );
+
+    expect(finalResponseInDb.id).toBe(initialResponseInDb.id);
+    expect(finalResponseInDb.request_id).toBe(initialResponseInDb.request_id);
+    expect(finalResponseInDb.user_id).toBe(initialResponseInDb.user_id);
+    expect(finalResponseInDb.status).toBe(initialResponseInDb.status);
+    expect(finalResponseInDb.content).toBe(updateData.content);
+  });
+});
+
+describe('Tests for deleting a request: DELETE /responses/:id', () => {
```

Comment:
these all have "request" left in the test names instead of "response"

Reply:
Done.